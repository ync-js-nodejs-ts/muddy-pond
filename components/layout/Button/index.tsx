import { ReactNode } from 'react'

interface IButtonProps {
  children: ReactNode
  color?: string
  callback?: () => void
  type?: 'button' | 'submit' | 'reset'
  tabIndex?: number
  className?: string
}

export const Button = ({ children, color, callback, type, tabIndex, className }: IButtonProps) => {
  return (
    <button
      tabIndex={tabIndex}
      type={type}
      onClick={callback}
      className={`border-2 rounded hover:bg-yellow-500 hover:text-white hover:transition duration-200 ease-in-out mt-2 min-w-min ${className || ''}`}
    >
      {children}
    </button>
  )
}
