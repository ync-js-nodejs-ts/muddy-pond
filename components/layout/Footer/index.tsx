import Image from 'next/image'
import { IconFacebook, IconInstagram, IconYoutube } from 'components/common/Icons'

export const Footer = () => (
  <footer className='w-full mt-10 h-28'>
    <div className=' bg-gray-900 h-full rounded-3xl rounded-b-none sm:h-auto pb-2 relative'>
      <div className='flex justify-center'>
        <figure className='bg-blue-800 px-3 pt-2 pb-1 rounded -mt-5'>
          <Image
            src='/img/logo.webp'
            alt='La Casa de mi Padre - Logo'
            width={65}
            height={36}
            priority
            className='unSelectElement'
            unselectable='on'
          />
        </figure>
      </div>

      <p className='text-white text-center'>Siguenos en</p>

      <section className='flex justify-center items-center mt-1'>
        <div className='mx-4'>
          <a href='https://www.youtube.com/channel/UCtttlnW0skxAug4MOo7CaeQ' target='_blank' rel='noopener noreferrer'>
            <IconYoutube width={30} height={30} />
          </a>
        </div>

        <div className='mx-4'>
          <a href='https://www.instagram.com/lcpcaracas' target='_blank' rel='noopener noreferrer'>
            <IconInstagram width={26} height={26} />
          </a>
        </div>

        <div className='mx-4'>
          <a href='https://www.facebook.com/lacasademipadrevenezuela' target='_blank' rel='noopener noreferrer'>
            <IconFacebook width={26} height={26} />
          </a>
        </div>
      </section>

      <div className='absolute bottom-2 right-4 text-yellow-400 md:hidden'>
        <small>Â©lcpcaracas</small>
      </div>
    </div>
  </footer>
)
