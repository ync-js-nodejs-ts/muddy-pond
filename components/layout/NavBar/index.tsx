import Image from 'next/image'
import Link from 'next/link'
import { useRouter } from 'next/router'
import { useState } from 'react'

import { IconHamburgerMenu } from 'components/common/Icons/HamburgerMenu'
import { DrawerNavigation } from './DrawerNavigation'

export const NavBar = () => {
  const [isOpen, setIsOpen] = useState(false)
  const router = useRouter()

  const routesLinkDesktop = [
    { label: 'Inicio', to: '/' },
    { label: 'Reservaciones', to: '/reservaciones' },
    { label: 'Blog', to: '/blog' },
    { label: 'Peticiones de oraciÃ³n', to: '/oracion' },
    { label: 'Contacto', to: '/contactanos' },
  ]

  const calculateLinkColor = () => {
    if (router.pathname === '/' ||router.pathname=== '/reservaciones' || router.pathname === '/oracion' || router.pathname === '/contactanos' ) return 'text-white'
    else return 'text-lcpBlue-200'
  }

  const handleIsOpenDrawer = () => setIsOpen((prevState) => !prevState)

  const calculateToFill = (route: string) => {
    if (route === '/reservaciones' || route === '/contactanos' || route === '/oracion' || route === '/') return 'white'
    else return '#1e478d'
  }

  return (
    <nav>
      <div className='max-w-screen-lg mx-auto flex justify-between px-5'>
        <section className='logo fixed top-0 left-3 z-20'>
          <figure className='bg-lcpBlue-300 px-3 pt-6 pb-2 cursor-pointer' onClick={handleIsOpenDrawer}>
            <Image className='pt-20' src='/img/logo.webp' alt='La Casa de mi Padre - Logo' width={65} height={36} priority />
          </figure>
        </section>

        <div className='fixed top-5 right-5 z-20 flex'>
          <div className={`items-center opacity-0 hidden transition-opacity duration-1000 ${isOpen ? 'lgmin:flex opacity-100' : ''}`}>
            {routesLinkDesktop.map((route) => (
              <Link href={route.to} key={route.to}>
                <a
                  onClick={() => setIsOpen(false)}
                  className={`border mr-2 border-solid border-yellow-500 w-40 py-2 rounded-2xl text-xs font-bold text-center ${calculateLinkColor()}`}
                >
                  {route.label}
                </a>
              </Link>
            ))}
          </div>

          <button onClick={handleIsOpenDrawer}>
            <IconHamburgerMenu height={28} width={28} stroke={calculateToFill(router.pathname)} />
          </button>
        </div>
      </div>

      <DrawerNavigation isOpen={isOpen} handleIsOpenDrawer={handleIsOpenDrawer} />
    </nav>
  )
}
