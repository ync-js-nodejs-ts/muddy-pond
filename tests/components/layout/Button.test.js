import { Button } from '../../../components/layout/Button'
import { shallow } from 'enzyme'

describe('Pruebas a <Button/>', () => {
  test('Debe de mostrarse correctamente', () => {
    const wrapper = shallow(<Button>Haz Click</Button>)
    expect(wrapper).toMatchSnapshot()
  })
})
