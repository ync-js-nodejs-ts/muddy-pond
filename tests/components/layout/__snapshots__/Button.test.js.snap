// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <Button/> Debe de mostrarse correctamente 1`] = `
<button
  className="border-2 rounded hover:bg-yellow-500 hover:text-white hover:transition duration-200 ease-in-out mt-2 min-w-min "
>
  Haz Click
</button>
`;
